### PR TITLE
refactor: migrate MyProfile to new MyAccount view

### PR DESCRIPTION
**Pull request by ghost**
Include the old MyProfile.vue in a subroute-friendly version (before it was a SfContentPage)

- [x] Added lang="ts" to every component related to MyProfile
- [x] Moved MyProfile-related components from components/* to pages/MyProfile/* 
- [x] Removed User interface from useUser and using Customer instead

### CODE REVIEW

File: packages/theme/modules/customer/pages/MyProfile/ProfileUpdateForm.vue [hunk 1]
```diff
@@ -0,0 +1,236 @@
+<template>
```
Comment (sethidden) [inline packages/theme/modules/customer/pages/MyProfile/ProfileUpdateForm.vue:1]: This is just a moved file. Not sure why is this appearing as a new file

File: packages/theme/modules/customer/pages/MyAccount/MyAccount.vue [hunk 2]
```diff
@@ -41,27 +41,30 @@
           </div>
         </div>
         <div class="sf-content-pages__content">
-          <router-view />
+          <router-view v-if="!$fetchState.pending" />
```
Comment (sethidden) [inline packages/theme/modules/customer/pages/MyAccount/MyAccount.vue:44]: There should be a SkeletonLeader next to here but I couldn't get it to show up during SSR. Maybe it's due to automatic Nuxt's lazy loading of subroutes

I don't know - I'll revisit it later before the merge to develop. I have a list of things to do before the merge

File: packages/theme/modules/customer/components/ProfileUpdateForm.vue [hunk 3]
```diff
@@ -1,237 +0,0 @@
-<template>
```
Comment (bartoszherba) [inline packages/theme/modules/customer/components/ProfileUpdateForm.vue:1]: Do we need this file? Its empty now.

File: packages/theme/modules/customer/pages/MyProfile/MyProfile.vue [hunk 4]
```diff
@@ -28,7 +28,7 @@
   </SfTabs>
 </template>
 
-<script>
+<script lang="ts">
```
Comment (bartoszherba) [inline packages/theme/modules/customer/pages/MyProfile/MyProfile.vue:31]: This is not about line31, but can we move MyProfile under MyAccount to reflect tree structure?

**Review by bartoszherba — CHANGES_REQUESTED** [review]

File: packages/theme/modules/customer/pages/MyProfile/MyProfile.vue (hunk 4 above)
Comment (sethidden) [reply packages/theme/modules/customer/pages/MyProfile/MyProfile.vue]: done

File: packages/theme/modules/customer/components/ProfileUpdateForm.vue (hunk 3 above)
Comment (sethidden) [reply packages/theme/modules/customer/components/ProfileUpdateForm.vue]: right, done - thanks

**Review by bartoszherba — APPROVED** [review]

**Pull request merged**